pkg/scheduler/plugins/deviceshare/deviceshare_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2019 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceshare
+
+import (
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"volcano.sh/volcano/pkg/scheduler/api"
+	"volcano.sh/volcano/pkg/scheduler/api/devices/nvidia/vgpu"
+	"volcano.sh/volcano/pkg/scheduler/framework"
+	"volcano.sh/volcano/pkg/scheduler/util"
+)
+
+const (
+	eps = 1e-8
+)
+
+func TestArguments(t *testing.T) {
+	framework.RegisterPluginBuilder(PluginName, New)
+	defer framework.CleanupPluginBuilders()
+
+	arguments := framework.Arguments{
+		"deviceshare.VGPUEnable":     true,
+		"deviceshare.SchedulePolicy": "binpack",
+		"deviceshare.ScheduleWeight": 10,
+	}
+
+	builder, ok := framework.GetPluginBuilder(PluginName)
+
+	if !ok {
+		t.Fatalf("should have plugin named %s", PluginName)
+	}
+
+	plugin := builder(arguments)
+	deviceshare, ok := plugin.(*deviceSharePlugin)
+
+	if !ok {
+		t.Fatalf("plugin should be %T, but not %T", deviceshare, plugin)
+	}
+
+	weight := deviceshare.scheduleWeight
+
+	if weight != 10 {
+		t.Errorf("weight should be 10, but not %v", weight)
+	}
+
+	if deviceshare.schedulePolicy != "binpack" {
+		t.Errorf("policy should be binpack, but not %s", deviceshare.schedulePolicy)
+	}
+}
+
+func addResource(resourceList v1.ResourceList, name v1.ResourceName, need string) {
+	resourceList[name] = resource.MustParse(need)
+}
+
+func TestVgpuScore(t *testing.T) {
+	gpuNode1 := vgpu.GPUDevices{
+		Name:   "node1",
+		Score:  float64(0),
+		Device: make(map[int]*vgpu.GPUDevice),
+	}
+	gpuNode1.Device[0] = vgpu.NewGPUDevice(0, 30000)
+	gpuNode1.Device[0].Type = "NVIDIA"
+	gpuNode1.Device[0].Number = 10
+	gpuNode1.Device[0].UsedNum = 1
+	gpuNode1.Device[0].UsedMem = 3000
+
+	gpunumber := v1.ResourceName("volcano.sh/vgpu-number")
+	gpumemory := v1.ResourceName("volcano.sh/vgpu-memory")
+
+	vgpu.VGPUEnable = true
+
+	p1 := util.BuildPod("c1", "p3", "", v1.PodPending, api.BuildResourceList("2", "10Gi"), "pg1", make(map[string]string), make(map[string]string))
+	addResource(p1.Spec.Containers[0].Resources.Requests, gpunumber, "1")
+	addResource(p1.Spec.Containers[0].Resources.Requests, gpumemory, "1000")
+	p1.Spec.Containers[0].Resources.Limits = make(v1.ResourceList)
+	addResource(p1.Spec.Containers[0].Resources.Limits, gpunumber, "1")
```

Comment:
`BuildResourceList` can directly add scaler resource.

Reply:
BuildResourceList doesn't add scalar resources to .containers.resources.limits, so we have to do it here manually